matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/task/TaskExecutor.kt

```diff
@@ -19,34 +19,30 @@ package im.vector.matrix.android.internal.task
 import im.vector.matrix.android.api.util.Cancelable
 import im.vector.matrix.android.internal.di.MatrixScope
 import im.vector.matrix.android.internal.extensions.foldToCallback
-import im.vector.matrix.android.internal.network.NetworkConnectivityChecker
 import im.vector.matrix.android.internal.util.MatrixCoroutineDispatchers
 import im.vector.matrix.android.internal.util.toCancelable
-import kotlinx.coroutines.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import timber.log.Timber
 import javax.inject.Inject
 import kotlin.coroutines.EmptyCoroutineContext
 
 @MatrixScope
-internal class TaskExecutor @Inject constructor(private val coroutineDispatchers: MatrixCoroutineDispatchers,
-                                                private val networkConnectivityChecker: NetworkConnectivityChecker) {
+internal class TaskExecutor @Inject constructor(private val coroutineDispatchers: MatrixCoroutineDispatchers) {
 
-    private val executorScope = CoroutineScope(SupervisorJob())
+    val executorScope = CoroutineScope(SupervisorJob())
 
     fun <PARAMS, RESULT> execute(task: ConfigurableTask<PARAMS, RESULT>): Cancelable {
         return executorScope
                 .launch(task.callbackThread.toDispatcher()) {
                     val resultOrFailure = runCatching {
                         withContext(task.executionThread.toDispatcher()) {
                             Timber.v("Enqueue task $task")
-                            retry(task.retryCount) {
-                                if (task.constraints.connectedToNetwork) {
-                                    Timber.v("Waiting network for $task")
-                                    networkConnectivityChecker.waitUntilConnected()
-                                }
-                                Timber.v("Execute task $task on ${Thread.currentThread().name}")
-                                task.execute(task.params)
-                            }
+                            Timber.v("Execute task $task on ${Thread.currentThread().name}")
```

Comment:
We do not wait for network anymore, what are the side effects?

Reply:
It's not reliable to wait for the network (same issue than for the sync), I prefer to have a retry policy on requests now. TaskExecutor will disappear in the future as it doesn't make sense anymore

Reply:
ok, thanks